fuse.go

```diff
@@ -222,35 +222,67 @@ func parseBuf(b []byte, alloc *Allocator, c *Conn) (Request, Response, error) {
 			return nil, nil, fmt.Errorf("fuse: read %d opcode %d but expected %d", len(b), h.opcode, h.len)
 		}
 	}
+	var handleErr error
 	switch h.opcode {
 	case OpInit:
-		return parseInit(b, alloc)
+		return parseInit(b, scope.alloc)
 	case OpStatfs:
-		return parseStatfs(b, alloc)
+		if req, resp, err = parseStatfs(b, scope.alloc); handler != nil && err == nil {
```

Comment:
Maybe redefine req and resp each time so we have the type info and don't need to cast?